hyppo/independence/friedmanRafsky.py

```diff
@@ -0,0 +1,297 @@
+import random
+from numba import jit
+
+import numpy as np
+
+from .base import IndependenceTest, IndependenceTestOutput
+
+
+class FriedmanRafsky(IndependenceTest):
+
+    r"""
+    Friedman-Rafksy (FR) test statistic and p-value.
+    This is a multivariate extension of the Wald-Wolfowitz
+    runs test for randomness. The normal concept of a 'run'
+    is replaced by a minimum spanning tree calculated between
+    the points in respective data sets with edge weights defined
+    as the Euclidean distance between two such points. After MST
+    has been determined, all edges such that both corresponding
+    nodes do not belong to the same class are severed and the
+    number of independent resulting trees is counted. This test is
+    consistent against similar tests
+    :footcite:p:`GSARMultivariateAdaptationofWaldWolfowitzTest`

```

Comment:
this is in `refs.bib` right?

Reply:
How would I reference a separate R package (seems like everything else is referencing articles)

Reply:
Well all I was asking is whether or not the reference you have here is in the bibtex file. As for citing R package, I would just look up how to cite packages in bibtex